grails-app/controllers/org/pih/warehouse/product/ProductController.groovy

```diff
@@ -271,7 +276,11 @@ class ProductController {
                     }
                 }
 
-                productInstance.validateRequiredFields()
+                if (location.isAccountingRequired()) {
+                    productInstance.validateRequiredFields()
+                } else {
+                    productInstance.validateRequiredFields([ProductField.GL_ACCOUNT])
```

Comment:
```suggestion
List<ProductField> ignoreProductFields = location.isAccountingRequired() ? [] : [ProductField.GL_ACCOUNT]
productInstance.validateRequiredFields(ignoreProductFields)
```

Reply:
this was my first version :D (look at the second commit) but I changed it because I thought it might look a bit better but I guess you guys disagree, I'll roll it back